modules/store/src/utils.ts

```diff
@@ -11,6 +11,79 @@ export function combineReducers<T, V extends Action = Action>(
   reducers: ActionReducerMap<T, V>,
   initialState?: Partial<T>
 ): ActionReducer<T, V>;
+/**
+ * @description
+ * Combines reducers for individual features into a single reducer.
+ *
+ * You can use this function to delegate handling of state transitions to multiple reducers, each acting on their
+ * own sub-state within the root state.
+ *
+ * @param reducers An object mapping keys of the root state to their corresponding feature reducer.
+ * @param initialState Provides a state value if the current state is `undefined`, as it is initially.
+ * @returns A reducer function.
+ *
+ * @usageNotes
+ *
+ * **Example combining two feature reducers into one "root" reducer**
+ *
+ * Feature A:
+ * ```ts
+ * export const featureKey = 'featureA';
+ * export interface State {
+ *   counterA: number;
+ * }
+ * const initialState: State = {
+ *   counterA: 0
+ * };
+ * export const featureAIncrement = createAction('[FeatureA] increment');
+ * export const reducer = createReducer(
+ *   initialState,
+ *   on(featureAIncrement, state => ({ counterA: state.counterA + 1 }))
+ * );
+ * ```
+ *
+ * Feature B:
+ * ```ts
+ * export const featureKey = 'featureB';
+ * export interface State {
+ *   counterB: number;
+ * }
+ * const initialState: State = {
+ *   counterB: 0
+ * };
+ * export const featureBIncrement = createAction('[FeatureB] increment');
+ * export const reducer = createReducer(
+ *   initialState,
+ *   on(featureBIncrement, state => ({ counterB: state.counterB + 1 }))
+ * );
+ * ```
+ *
+ * Combining the feature reducers using `combineReducers`:
+ * ```ts
+ * import * as fromFeatureA from './featureA';
+ * import * as fromFeatureB from './featureB';
+ *
+ * export interface State {
+ *   [fromFeatureA.featureKey]: fromFeatureA.State;
+ *   [fromFeatureB.featureKey]: fromFeatureB.State;
+ * }
+ * export const reducer = combineReducers({
+ *   [fromFeatureA.featureKey]: fromFeatureA.reducer,
+ *   [fromFeatureB.featureKey]: fromFeatureB.reducer
+ * });
+ * ```
+ *
+ * You can also override the initial states of the sub-features:
+ * ```ts
+ * export const reducer = combineReducers({
+ *   [fromFeatureA.featureKey]: fromFeatureA.reducer,
+ *   [fromFeatureB.featureKey]: fromFeatureB.reducer
+ * }, {
+ *   [fromFeatureA.featureKey]: { counterA: 13 },
+ *   [fromFeatureB.featureKey]: { counterB: 37 }
+ * });
```

Comment:
Personally, I think this would be enough to add in the example notes.
Maybe also add in what state tree it results?

```json
{
   featureA: { counterA: 13 },
   featureB: { counterB: 37 }
}
```

What do you think?

Reply:
I think that could be enough if you think that it's obvious that `fromFeatureA` is imported as `import * as fromFeatureA from "./feature_a.ts";`.

I guess in that case you could also get rid of the computed object keys and just use `{ featureA: fromFeatureA.reducer /* , ... */ }` or even just `{ featureA: featureAReducer /* , ... */ }` and do the same for the initial state. That would also make your example state tree redundant.

Reply:
I like it, that's probably better.
The main idea is the just trim the example code to what is important for `combineReducers`.

Reply:
@cmd-johnson I'm just checking in to see if you want to make these changes.

Reply:
@timdeschryver sorry, I totally forgot about this one. I've pushed the shortened example, what do you think?

Reply:
Np! I like the changes, thanks!